### ISSUE TITLE
How to create an async Future from the main thread?

### ISSUE THREAD

**Issue by alexanderjarvis** [Thomvis/BrightFutures#116]
Hey great work by the way! I've just hit a stumbling block however and would love to get some help for something that might be a typical use case too.

I have extended Alamofire to return type `Future<JSON, NSError>` and have a service that calls this code (from the main thread).

However I discovered that all the map functions I use on that `Future` execute on the main thread unless if I specify a different context (this is not desired because it seems repetitive).

So basically my question is how do you call a function that returns a `Future` in a way that it will be executed on the global queue? (The `Future` itself is created with a `Promise`).

I've tried looking at the future function and it's different variations but haven't got anything to work yet.

**Comment by Thomvis** [comment]
Thanks for your kind words.

Firstly, as you probably already know, all callbacks work according to the default threading model ([see README](https://github.com/Thomvis/BrightFutures#default-threading-model)). This basically means that callbacks that are added to a future on the main thread, will also execute on the main thread. This seems to work pretty well in de context of app development. But of course, there are exceptions.

In the case where you want to have some callbacks execute off the main thread (e.g. before calling onSuccess on the main thread) you'll indeed have to specify the explicit execution context (either the global context or the immediate execution context). I personally think the syntax looks quite nice and, albeit repetitive, clear.

There has been some discussion about making the threading model configurable. You can read it here: https://github.com/Thomvis/BrightFutures/pull/109. My conclusion was that it is not worth it, but feel free to reopen the discussion.

**Comment by alexanderjarvis** [comment]
Hi Thomas,

Sorry I was extremely slow to reply on this issue.

Basically what I think could be done to improve your Futures library is to allow the execution context to be passed as a parameter to the Promise when it is created. This then means that from the main thread a library can setup a promise where by default the map function operates on the global queue.

This makes sense for the use case of taking a HTTP response and transforming it at different stages (to JSON, then to some intermediate representation and then to the local model, which you want to be taking place off the main thread).

I think this is a separate concern than the one in #109 where you talk about a global setting for a threading model, but please correct me.
